src/tabs/__tests__/tabs.test.tsx

```diff
@@ -752,6 +796,272 @@ describe('Tabs', () => {
+
+              return item;
+            })}
+            activeTabId={firstTabId}
+            onChange={() => void 0}
+          />
+        );
+
+        fireEvent.mouseEnter(wrapper.findTabLinkById('second')!.getElement());
+
+        expect(wrapper.findTabLinkById('second')!.findDisabledReason()!.getElement()).toHaveTextContent(
+          'disabled reason'
+        );
+
+        fireEvent.mouseLeave(wrapper.findTabLinkById('second')!.getElement());
+
+        expect(wrapper.findTabLinkById('second')!.findDisabledReason()).toBe(null);
+      });
+
+      test('has no aria-describedby by default', () => {
+        const firstTabId = defaultTabs[0].id;
+        const { wrapper } = renderTabs(<Tabs tabs={defaultTabs} activeTabId={firstTabId} onChange={() => void 0} />);
+
+        expect(wrapper.findTabLinkById('second')!.getElement()).not.toHaveAttribute('aria-describedby');
+      });
+
+      test('has no aria-describedby without disabledReason', () => {
+        const firstTabId = defaultTabs[0].id;
+        const { wrapper } = renderTabs(
+          <Tabs
+            tabs={defaultTabs.map(item => {
+              if (item.id === 'second') {
+                return {
+                  ...item,
+                  disabled: true,
+                };
+              }
+
+              return item;
+            })}
+            activeTabId={firstTabId}
+            onChange={() => void 0}
+          />
+        );
+
+        expect(wrapper.findTabLinkById('second')!.getElement()).not.toHaveAttribute('aria-describedby');
+      });
+
+      test('has aria-describedby with disabledReason', () => {
+        const firstTabId = defaultTabs[0].id;
+        const { wrapper } = renderTabs(
+          <Tabs
+            tabs={defaultTabs.map(item => {
+              if (item.id === 'second') {
+                return {
+                  ...item,
+                  disabled: true,
+                  disabledReason: 'disabled reason',
+                };
+              }
+
+              return item;
+            })}
+            activeTabId={firstTabId}
+            onChange={() => void 0}
+          />
+        );
+
+        expect(wrapper.findTabLinkById('second')!.getElement()).toHaveAttribute('aria-describedby');
+      });
+
+      test('has hidden element (linked to aria-describedby) with disabledReason', () => {
+        const firstTabId = defaultTabs[0].id;
+        const { wrapper } = renderTabs(
+          <Tabs
+            tabs={defaultTabs.map(item => {
+              if (item.id === 'second') {
+                return {
+                  ...item,
+                  disabled: true,
+                  disabledReason: 'disabled reason',
+                };
+              }
+
+              return item;
+            })}
+            activeTabId={firstTabId}
+            onChange={() => void 0}
+          />
+        );
+
+        expect(wrapper.findTabLinkById('second')!.find('span[hidden]')!.getElement()).toHaveTextContent(
```

Comment:
Would it make sense to merge this test with previous one?

Instead of checking the presence of aria-describedby you can do:
```
const describedBy = wrapper.findTabLinkById('second').getElement().getAttribute('aria-describedby');
const description = wrapper.findTabLinkById('second').find(`#${descrbedBy}`).getElement().textContent;
```

Reply:
Makes sense. Updated

Reply:
I don't see this change in code, could you share a link to the line please?

Reply:
Sure, here you go:

https://github.com/cloudscape-design/components/pull/2413/files#diff-fc85e49981e81cc2dbe0ff658349f22e0fdb03d247a6aa3370b4fd828c8cbb6eR1133

But there are more test suites for these a11y attributes above which could be combined.

Reply:
Ah, so what I meant is that not only the tests can be merged but you can also ensure that the describedby points to the correct element. See code above

Reply:
now I got it! tbh, I don't think it makes much sense, because the purpose of this test is to check presence of the hidden element with certain content, and the way how it's associated with the id is handled by `useHiddenDescription` hook, so this case is rather internal implementation of it

Reply:
I would not agree that it is an internal implementation because the only presence of the hidden element does not ensure this information is announced. The element must be connected with the aria-describedby. The hook helps with establishing that connection but does not guarantee that (e.g. if you don't pass the ID to the correct element or accidentally override it with another ID).

Reply:
updated. introduced `data-describedby-id`, because requesting elements via `.find('#some-id')` is not a valid way 

> accidentally override it with another ID
I would not consider it, because the dom element representing the description is encapsulated within `useHiddenDescription` hook, so there is not accidental way to override it

Reply:
Then you can use `.find('[id="your-id"]')`

Reply:
Didn't know it was possible :)
Updated